### ISSUE TITLE
Handle multi_class="multinomial" sklearn deprecation

### ISSUE THREAD

**Issue by Valentin-Laurent** [scikit-learn-contrib/MAPIE#615]
The default estimator for MapieClassifier is `LogisticRegression(multi_class="multinomial")`. This parameter is deprecated by sklearn and will be removed in 1.7.

- With n_class >= 3, this will have no impact (multinomial is chosen by default)
- With n_class = 2 (allowed in MAPIE, only with LAC conformity score), it will not be possible to chose `multinomial`. There is a discussion about the loss of this functionality in [the sklearn deprecation PR](https://github.com/scikit-learn/scikit-learn/pull/28703).

We need to discuss how we want to handle this. IMO, the decision tree looks like this:

> Do we want to support binary classification in v1?
> - No: we raise a warning/error about it in v1, and we remove `multi_class="multinomial"` everywhere in the code
> - Yes: do we really need `multinomial` for binary classification? Again, see the interesting discussion in [the sklearn deprecation PR](https://github.com/scikit-learn/scikit-learn/pull/28703).
>   - Yes: we should get in touch with sklearn core dev to discuss this need
>   - No: we remove `multi_class="multinomial"` everywhere in the code

**Comment by Valentin-Laurent** [comment]
Decision: we just remove `multi_class="multinomial"` everywhere in the code.

**Comment by FaustinPulveric** [comment]
Removing `multi_class="multinomial"` has led to a regression in binary calibrations. For now, the test `test_correct_results_binary` has been bypassed by hardcoding `results_binary["split"]` (see details in commit 6a39210).
